Apps/Contoso.WPF.Demo/App.xaml.cs

```diff
@@ -13,10 +14,84 @@ namespace Contoso.WPF.Demo
     /// </summary>
     public partial class App
     {
+        private const string LogTag = "AppCenterDemo";
+
         protected override void OnStartup(StartupEventArgs e)
         {
             AppCenter.LogLevel = LogLevel.Verbose;
+            Crashes.ShouldAwaitUserConfirmation = ConfirmationHandler;
+            Crashes.ShouldProcessErrorReport = ShouldProcess;
+
+            // Event handlers.
+            Crashes.SendingErrorReport += SendingErrorReportHandler;
+            Crashes.SentErrorReport += SentErrorReportHandler;
+            Crashes.FailedToSendErrorReport += FailedToSendErrorReportHandler;
             AppCenter.Start("f4e2a83d-3052-4884-8176-8b2c50277d16", typeof(Analytics), typeof(Crashes));
+            Crashes.HasCrashedInLastSessionAsync().ContinueWith(hasCrashed =>
+            {
+                AppCenterLog.Info(LogTag, "Crashes.HasCrashedInLastSession=" + hasCrashed.Result);
+            });
+            Crashes.GetLastSessionCrashReportAsync().ContinueWith(task =>
+            {
+                AppCenterLog.Info(LogTag, "Crashes.LastSessionCrashReport.Exception=" + task.Result?.Exception);
+            });
+        }
+
+        bool ShouldProcess(ErrorReport report)
+        {
+            AppCenterLog.Info(LogTag, "Determining whether to process error report");
+            return true;
+        }
+
+        bool ConfirmationHandler()
+        {
+            Current.Dispatcher.Invoke(() =>
+            {
+                var dialog = new UserConfirmationDialog();
+                if (dialog.ShowDialog() ?? false)
+                {
+                    Crashes.NotifyUserConfirmation(dialog.ClickResult);
+                }
+            });
+            return true;
+        }
+
+        static void SendingErrorReportHandler(object sender, SendingErrorReportEventArgs e)
+        {
+            AppCenterLog.Info(LogTag, "Sending error report");
+            var report = e.Report;
+            if (report.Exception != null)
+            {
+                AppCenterLog.Info(LogTag, report.Exception.ToString());
```

Comment:
Doesn't seem `report.Exception.ToString()` is useful. Please log at least error ID.